File: algorithms/codefights/interview-practice/arrays/rotate-image/rotate-image.js
function rotateImage(a) {
    const l = a.length;

    for (let i = 0; i < l / 2; i++) {
        for (let j = i; j < l - i - 1; j++) {
            let tmp = a[i][j];
            a[i][j] = a[l - j - 1][i];
            a[l - j - 1][i] = a[l - i - 1][l -j - 1];
            a[l - i - 1][l -j - 1] = a[j][l - i - 1];
            a[j][l - i - 1] = tmp;
        }
    }

    return a;
}

File: leetcode/048-rotate-image/rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        size_t n = matrix.size();
        for (size_t i = 0; i < n / 2; i++) {
            for (size_t j = i; j < n - i - 1; j++) {
                int tmp = matrix[i][j];
                matrix[i][j] = matrix[n - j - 1][i];
                matrix[n - j - 1][i] = matrix[n - i - 1][n - j - 1];
                matrix[n - i - 1][n - j - 1] = matrix[j][n - i - 1];
                matrix[j][n - i - 1] = tmp;
            }
        }
    }
};
